dijet/tasks/correlated_fit.py

```diff
@@ -0,0 +1,89 @@
+# coding: utf-8
+
+import law
+from columnflow.util import maybe_import
+
+from dijet.tasks.util import chi2
+
+sc = maybe_import("scipy.optimize")
+nd = maybe_import("numdifftools")
+np = maybe_import("numpy")
+
+class CorrelatedFit():
+    """
+    Task to calculated correlated fit to widths.
+
+    ## Covariance matrix 
+    [1] Based on K. Goebel dissertation ch. A.3.
+    https://www.physik.uni-hamburg.de/en/iexp/gruppe-haller/scientific-output/documents/thesis-kristin-goebel.pdf
+
+    Example code here
+    https://github.com/UHH2/DiJetJERC/blob/ff98eebbd44931beb016c36327ab174fdf11a83f/
+    JERSF_Analysis/JER/wide_eta_binning/functions.C#L558
+    func. createCov()
+    """
+
+    def createCov(self, widths, widths_err):
+        widths_err2 = pow(widths_err, 2)  # square each element
+
+        # 3x3 matrix with the smaller error of indices i&j (Check [1])
+        # [1,2,3] ->
+        # [
+        #  [1,1,1]
+        #  [1,2,2]
+        #  [1,2,3]
+        # ]
+        matrix_err = np.maximum.outer(widths_err2,widths_err2)
+
+        ratio = ( pow(widths,2) / (2 * widths_err2) )
+        n_ratio = pow((ratio[:, None] / ratio),2)  # get ij element
+        w_ratio = (widths[:, None] / widths)  # get ij element
+
+        # Calculate the upper triangle of the covariance matrix.
+        y_cov_mc = np.triu(matrix_err * w_ratio * n_ratio)
+
+        # Fill in the lower triangle of the covariance matrix.
+        y_cov_mc += y_cov_mc.T - np.diag(y_cov_mc.diagonal())
+
+        return np.nan_to_num(y_cov_mc)
+
+    def correlated_fit(self, wmax, data, cov):
```

Comment:
This doesn't depend on `self` -> please implement outside class (recommended) or mark as `@staticmethod`
```suggestion
    @staticmethod
    def correlated_fit(wmax, data, cov):
```